balboa-agent/src/main/scala/com/socrata/balboa/agent/MetricConsumer.scala

```diff
@@ -0,0 +1,280 @@
+
+/**
+  * The MetricConsumer consumes metrics from data files from within a specific directory.  Any metrics extracted will
+  * be pushed to a specified [[com.socrata.metrics.MetricQueue]].
+  *
+  * <br>
+  *
+  * Given a root directory, the MetricConsumer will recursively extract all the directories under the root directory.
+  * For each directory, files are processed in reverse alphabetical order excluding the last file in the order.  It is
+  * common and suggested that metric producers use timestamps in their file names and not place any additional files
+  * within the same directory they write metrics to.
+  *
+  */
+//  TODO: Make this contract better (assuming last file off-limits is dumb)
+
+object MetricConsumer {
+  val TIMESTAMP: String = "timestamp"
+  val ENTITY_ID: String = "entityId"
+  val NAME: String = "name"
+  val VALUE: String = "value"
+  val RECORD_TYPE: String = "type"
+  val fields: util.List[String] = util.Arrays.asList(TIMESTAMP, ENTITY_ID, NAME, VALUE, RECORD_TYPE)
+  val integerPattern: Pattern = Pattern.compile("-?[0-9]+")
+}
+
+/**
+  * Creates a MetricConsumer that processes files from `directory` and emits them to `metricPublisher`.  The `fileProvider`
+  * is a control mechanism that allows clients to make determinations on which [[java.io.File]]s can be processed.
+  *
+  * @param directory       Directory in which to process metrics.
+  * @param metricPublisher Queue to emit metrics to.
+  * @param fileProvider    The { @link FileFilter} used to determine which files are allowed to be processed.
+  */
+class MetricConsumer(val directory: File, val metricPublisher: MetricQueue, val fileProvider: MetricFileProvider)
+  extends Runnable with AutoCloseable with LazyLogging {
+
+  if (directory == null || !directory.isDirectory) throw new IllegalArgumentException("Illegal Data directory " + directory)
+  if (metricPublisher == null) throw new NullPointerException("Metric Queue cannot be null")
+
+  /**
+    * Creates Metric consumer that will attempt to find all the metric data within a directory
+    * and push them to a metricPublisher. Does not take responsibility for closing the metricPublisher.
+    */
+  def this(directory: File, metricPublisher: MetricQueue) {
+    this(directory, metricPublisher, AlphabeticMetricFileProvider(directory.toPath))
+  }
+
+  /**
+    * Attempts to process all the sub directories on the root directories for all possible
+    * metrics.
+    */
+  def run() {
+    logger.info("Looking for metrics files recursively in '{}'", this.directory.getAbsolutePath)
+    val runTimer: Timer.Context = BalboaAgentMetrics.totalRuntime.time
+    val start: Long = System.currentTimeMillis
+    var recordsProcessed: Int = 0
+    // Treat each individual Metric data file as its own isolated run.
+    // We are trying to prevent the failure to process one file from blocking or preventing the processing
+    // of others.
+    fileProvider.provide.foreach { metricsEventLog =>
```

Comment:
Yeah, plan was to migrate to processing in one pass instead of this piece-meal approach.